src/components/AnalysisPage/ThicknessAnalysisBox/indexedDBBlobStorage.js

```diff
@@ -0,0 +1,89 @@
+    const request = indexedDB.open("PlyBlobDatabase", 1);
+    request.onupgradeneeded = function () {
+      const db = request.result;
+      if (!db.objectStoreNames.contains("FilesStore")) {
+        db.createObjectStore("FilesStore", { keyPath: "id" });
+      }
+    };
+    request.onerror = function (event) {
+      console.error("Database error:", event.target.errorCode);
+      reject("Database error: " + event.target.errorCode);
+    };
+    request.onsuccess = function () {
+      resolve(request.result);
+    };
+  });
+};
+
+export const storeBlob = async (partId, units, thresholdValue, blob) => {
+  console.log("Storing blob with key:", `${partId}-${units}-${thresholdValue}`);
+  try {
+    const db = await openDatabase();
+    const transaction = db.transaction("FilesStore", "readwrite");
+    const store = transaction.objectStore("FilesStore");
+    const key = `${partId}-${units}-${thresholdValue}`;
+
+    const request = store.put({ id: key, blob: blob });
+    return new Promise((resolve, reject) => {
+      request.onerror = function () {
+        console.error("Error storing blob:", request.error);
+        reject("Failed to store blob");
+      };
+      request.onsuccess = function () {
+        console.log("Blob stored successfully with key:", key);
+        resolve(request.result);
+      };
+      transaction.oncomplete = function () {
+        db.close();
+      };
+      transaction.onerror = function (event) {
+        console.error("Transaction failed:", event);
+      };
+    });
+  } catch (error) {
+    console.error("Error during storage operation:", error);
+    throw error;
+  }
+};
+
+export const retrieveBlob = async (partId, units, thresholdValue) => {
+  console.log(
+    "Retrieving blob with key:",
+    `${partId}-${units}-${thresholdValue}`
+  );
+  try {
+    const db = await openDatabase();
+    const transaction = db.transaction("FilesStore", "readonly");
+    const store = transaction.objectStore("FilesStore");
+    const key = `${partId}-${units}-${thresholdValue}`;
+    const request = store.get(key);
+
+    return new Promise((resolve, reject) => {
+      request.onerror = function (event) {
+        console.error("Error retrieving blob:", event.target.error);
+        reject("Could not retrieve the file: " + event.target.errorCode);
+      };
+      request.onsuccess = function () {
+        if (request.result) {
+          console.log("Blob retrieved successfully with key:", key);
+          resolve(request.result.blob);
+        } else {
+          console.log("No blob found for key:", key);
+          resolve(null); // or reject if you prefer to treat no data as an error
+        }
+      };
+      transaction.oncomplete = function () {
+        db.close();
+      };
+      transaction.onerror = function (event) {
+        console.error("Transaction failed:", event);
+      };
+    });
+  } catch (error) {
+    console.error("Error during retrieval operation:", error);
+    throw error;
+  }
```

Comment:
unbelievable, you're right again. 

But I'm reading that we don't even need to throw the error because it is redundant after rejecting the promise. 
Does that sound accurate?

So i'm just going to remove the try-catch from both storeBlob and retrieveBlob.

Reply:
Its not just redundant for the rejected promise, the rejection never actually goes into the catch block.

For example if you run this code:
```js
function testPromiseBehavior() {

	try {
		return new Promise((resolve, reject) => {
			console.log('1')
			return reject('bad thing')
		})
	} catch(err) {
		console.log('2')
	}
}

testPromiseBehavior()
```

It only logs 1 and doesn't log 2

BUT for the other stuff above the 'return new Promise' statement; if anything in there throws an error then it will still go into the catch block.
See this example:
```
async function testPromiseBehavior() {

	try {

		doOtherThing()

		return new Promise((resolve, reject) => {
			console.log('1')
			return reject('bad thing')
		})
	} catch(err) {
		console.log('2')
	}
}

testPromiseBehavior()

function doOtherThing() {
	throw new Error('a bad thing from doOtherThing')
}

```
This will log 2.


And also if you don't already know; a catch block with just rethrowing is totally redundant. It doesn't do anything.

```js
try {
  doThingThatMightError()
} catch(err) {
  // this whole catch statement does nothing and the try catch should just be removed
  throw err
}
```

@ErikAmerico 
Let me know if anything here doesn't make sense